assets/src/components/stacks/state/StackStateGraph.tsx

```diff
@@ -9,55 +9,96 @@ import { ReactFlowGraph } from '../../utils/reactflow/ReactFlowGraph'
 
 import { LayoutOptions } from 'elkjs'
 import { StackStateGraphNode } from './StackStateGraphNode'
+import { Flex, Input, SearchIcon } from '@pluralsh/design-system'
+import Fuse from 'fuse.js'
+import { isEmpty } from 'lodash'
 
 const nodeTypes = {
   [NodeType.Stage]: StackStateGraphNode,
 }
 
-function getNodesAndEdges(state: StackState) {
+const searchOptions = {
+  keys: [
+    'name',
+    {
+      name: 'configuration',
+      getFn: (r) => JSON.stringify(r?.configuration ?? {}),
+    },
+  ],
+  threshold: 0.25,
+}
+
+function getNodesAndEdges(state: StackState, query: string) {
+  const resources = state.state?.filter(isNonNullable) ?? []
+
+  const filteredIds = isEmpty(query)
+    ? new Set(resources.map(({ identifier }) => identifier))
+    : new Set(
+        new Fuse(resources, searchOptions)
+          .search(query)
+          .map(({ item }) => [item.identifier, ...(item.links ?? [])])
+          .flat()
```

Comment:
Similar to the cluster network graph this add not only items found by the search but also items linked directly to them.